frontend/public/components/utils/resource-icon.tsx

```diff
@@ -2,23 +2,36 @@ import * as React from 'react';
 import * as classNames from 'classnames';
 import * as _ from 'lodash-es';
 
-import { K8sResourceKindReference } from '../../module/k8s';
+import {
+  K8sGroupVersionKind,
+  K8sResourceKindReference,
+  referenceForGroupVersionKind,
+} from '../../module/k8s';
 import { modelFor } from '../../module/k8s/k8s-models';
 import { kindToAbbr } from '../../module/k8s/get-resources';
 
 const MEMO = {};
 
-export const ResourceIcon: React.SFC<ResourceIconProps> = ({ className, kind }) => {
-  // if no kind, return null so an empty icon isn't rendered
-  if (!kind) {
+export const ResourceIcon: React.SFC<ResourceIconProps> = ({
+  className,
+  groupVersionKind,
+  kind,
+}) => {
+  // if no kind or groupVersionKind, return null so an empty icon isn't rendered
+  if (!kind && !groupVersionKind) {
     return null;
   }
-  const memoKey = className ? `${kind}/${className}` : kind;
+  const kindReference =
+    kind ||
+    referenceForGroupVersionKind(groupVersionKind.group)(groupVersionKind.version)(
+      groupVersionKind.kind,
+    );
```

Comment:
Um... why do we have the other one? 😕

Reply:
that was there and being used across, so didn't remove that. We could have marked `referenceForGroupVersionKind` as deprecated in favor of `getReference` but `getReference` had a cleaner signature and is already deprecated. 
Added comment as eventually, we'll move `resource-icon` to SDK then we have it.

Reply:
I don't understand the logic behind the new function's existence (I probably approved this so this is more of a critique of my 2021 self lol). If we didn't want to update the bajillion cases to use the direct call methodology, why diverge the code by duplicating the implementation? Just call the SDK function, or at least deprecate the old one. We should do this part of this PR now because it was not done before.

eg, Call the SDK function:
```
// TODO(alecmerdler): Replace all manual string building with this function
export const referenceForGroupVersionKind = (group: string) => (version: string) => (
  kind: string,
) => getReference(group, version, kind);
```